### PR TITLE
Add restart --cidfile, --filter

### PR DESCRIPTION
**Pull request by sstosh**
--cidfile : Read container ID from the specified file and restart the container.
--filter : restart the filtered container.

Signed-off-by: Toshiki Sonoda <sonoda.toshiki@fujitsu.com>

#### Does this PR introduce a user-facing change?

```release-note
User can use podman restart --cidfile, --filter option.
```

### CODE REVIEW

**Review by ashley-cui — COMMENTED** [review]
Greatt job @sstosh! LGTM!

File: docs/source/markdown/podman-restart.1.md [hunk 1]
```diff
@@ -14,14 +14,46 @@ Containers will be stopped if they are running and then restarted. Stopped
...
 
 ## OPTIONS
+
 #### **--all**, **-a**
+
 Restart all containers regardless of their current state.
 
+#### **--cidfile**
+
+Read container ID from the specified file and pause the container.  Can be specified multiple times.
```
Comment (rhatdan) [inline docs/source/markdown/podman-restart.1.md:24]: Pause?

**Review by rhatdan — APPROVED** [review]
LGTM except for cut/paste error.

File: docs/source/markdown/podman-restart.1.md [hunk 2]
```diff
@@ -14,14 +14,46 @@ Containers will be stopped if they are running and then restarted. Stopped
...
+| label           | [Key] or [Key=Value] Label assigned to a container                               |
+| exited          | [Int] Container's exit code                                                      |
+| status          | [Status] Container's status: 'created', 'exited', 'paused', 'running', 'unknown' |
+| ancestor        | [ImageName] Image or descendant used to create container                         |
+| before          | [ID] or [Name] Containers created before this container                          |
+| since           | [ID] or [Name] Containers created since this container                           |
+| volume          | [VolumeName] or [MountpointDestination] Volume mounted in container              |
+| health          | [Status] healthy or unhealthy                                                    |
+| pod             | [Pod] name or full or partial ID of pod                                          |
+| network         | [Network] name or full ID of network                                             |
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-restart.1.md:48]: I think it makes sense to have the id first, and maybe the name, but the rest I'd put into alpha order.

File: cmd/podman/containers/restart.go [hunk 3]
```diff
@@ -47,20 +48,35 @@ var (
...
+	restartTimeout  uint
 )
 
 func restartFlags(cmd *cobra.Command) {
 	flags := cmd.Flags()
 
-	flags.BoolVarP(&restartOptions.All, "all", "a", false, "Restart all non-running containers")
-	flags.BoolVar(&restartOptions.Running, "running", false, "Restart only running containers when --all is used")
+	flags.BoolVarP(&restartOpts.All, "all", "a", false, "Restart all non-running containers")
+	flags.BoolVar(&restartOpts.Running, "running", false, "Restart only running containers when --all is used")
```
Comment (TomSweeneyRedHat) [inline cmd/podman/containers/restart.go:62]: Maybe it's my end of week brain not clicking, but I don't see a requirement that the `--all` option is needed with the `running` option?  Is it required?  If so, the man page needs to reflect that.  However, I think it's not required, so I would simply change this to:

```suggestion
	flags.BoolVar(&restartOpts.Running, "running", false, "Restart only running containers")
```

File: docs/source/markdown/podman-restart.1.md (hunk 2 above)
Comment (rhatdan) [reply docs/source/markdown/podman-restart.1.md:48]: I think this should be done in @edsantiago effort to fix up cut and pastes in man pages.  I believe this section is repeated?

**Review by vrothberg — COMMENTED** [review]
@sstosh could you rebase?

LGTM

@containers/podman-maintainers PTAL

**Pull request merged**